eel/edge.py

```diff
@@ -1,20 +1,48 @@
-import platform
+import os
 import subprocess as sps
 import sys
 from typing import List
 
 from eel.types import OptionsDictT
 
-name: str = 'Edge'
+name: str = "Edge"
 
 
-def run(_path: str, options: OptionsDictT, start_urls: List[str]) -> None:
-    cmd = 'start microsoft-edge:{}'.format(start_urls[0])
-    sps.Popen(cmd, stdout=sys.stdout, stderr=sys.stderr, stdin=sps.PIPE, shell=True)
+def run(path: str, options: OptionsDictT, start_urls: List[str]) -> None:
+    if path != "edge_legacy":
+        if options["app_mode"]:
+            for url in start_urls:
+                sps.Popen([path, "--app=%s" % url] + options["cmdline_args"], stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+        else:
+            args = options["cmdline_args"] + start_urls
+            sps.Popen([path, "--new-window"] + args, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+    else:
+        cmd = "start microsoft-edge:{}".format(start_urls[0])
+        sps.Popen(cmd, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE, shell=True)
 
 
 def find_path() -> bool:
-    if platform.system() == 'Windows':
-        return True
+    if sys.platform in ["win32", "win64"]:
+        return _find_edge_win()
+    else:
+        return None
 
-    return False
+
+def _find_edge_win():
+    import winreg as reg
+
+    reg_path = r"SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\msedge.exe"
+
+    for install_type in reg.HKEY_CURRENT_USER, reg.HKEY_LOCAL_MACHINE:
+        try:
+            reg_key = reg.OpenKey(install_type, reg_path, 0, reg.KEY_READ)
+            edge_path = reg.QueryValue(reg_key, None)
+            reg_key.Close()
+            if not os.path.isfile(edge_path):
+                continue
+        except WindowsError:
+            edge_path = "edge_legacy"
+        else:
+            break
+
+    return edge_path
```

Comment:
I see two potential issues with this loop:

1.  `edge_path` here is potentially unbound (think what happens if for God knows what reason no WindowsError is ever raised *and* Edge not found), and its type is also technically indeterminate, which is a bad situation.

    To address this, I'd suggest setting `edge_path: str = 'start microsoft-edge'` as a fallback value before entering the loop.

2. Your loop runs exactly twice, once for `reg.HKEY_CURRENT_USER` and once for `reg.HKEY_LOCAL_MACHINE`. These options are fixed and there is no reason we should think that further iterations should become available in the future, so the loop can be unrolled.

    I would take the point that it's rather ugly to write this twice in a row, so what I would do is add a separate function to query the registry, e.g. `_query_winreg(key: str, sub_key: str) -> Optional[str]` which implements the checking logic, and returns either the string if `os.path.isfile(edge_path)` is True and no WindowsError (for sub_key not found) is raised.
    You could then set just do
    ```
    edge_path = _query_winreg(reg.HKEY_LOCAL_MACHINE, reg_path) or edge_path
    return _query_winreg(reg.HKEY_CURRENT_USER, reg_path) or edge_path
    ```
    which would make this quite clean and more readable in my opinion (though one can argue about hijacking `or` to select between truthy and falsy *values*.